ext/standard/tests/general_functions/get_debug_type_basic.phpt

```diff
@@ -0,0 +1,43 @@
+--TEST--
+Test get_debug_type() class reading
+--FILE--
+<?php
+
+namespace Demo {
+    class ClassInNamespace {
+
+    }
+}
+
+namespace {
+    class ClassInGlobal {
+
+    }
+
+    /* tests against an object type */
+    echo get_debug_type(new ClassInGlobal()) . "\n";
+    echo get_debug_type(new Demo\ClassInNamespace()) . "\n";
+
+    /* scalars */
+    echo get_debug_type("foo") . "\n";
+    echo get_debug_type(false) . "\n";
+    echo get_debug_type(true) . "\n";
+    echo get_debug_type(1) . "\n";
+    echo get_debug_type(1.1) . "\n";
+    echo get_debug_type([]) . "\n";
+    echo get_debug_type(null) . "\n";
+    echo get_debug_type(fopen(__FILE__, 'r')) . "\n";
```

Comment:
I'd suggest adding a test case with a closed resource also:
```php
fclose($h = fopen(__FILE__, 'r'));
echo get_debug_type($h) . "\n";
```

Reply:
I wonder if it makes sense to include the resource type in the resulting string, eg `resource (stream-context)`: https://3v4l.org/cD69T ?

Reply:
I have added this in but will likely need to include it as a separate part of the vote.

Reply:
Sub-votes should be exceptional to me. If there are any questions, they should be resolved before opening the vote.